contracts/token/ERC721/extensions/ERC721Wrapper.sol

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "../ERC721.sol";
+import "../utils/ERC721Holder.sol";
+
+/**
+ * @dev Extension of the ERC721 token contract to support token wrapping.
+ *
+ * Users can deposit and withdraw an "underlying token" and receive a "wrapped token" with a matching tokenId. This is useful
+ * in conjunction with other modules. For example, combining this wrapping mechanism with {ERC721Votes} will allow the
+ * wrapping of an existing "basic" ERC721 into a governance token.
+ *
+ * _Available since v4.9.0_
+ */
+abstract contract ERC721Wrapper is ERC721, ERC721Holder {
+    IERC721 private immutable _underlying;
+
+    // Kept as bytes12 so it can be packed with an address
+    // Equal to 0xb125e89df18e2ceac5fd2fa8
+    bytes12 private constant _WRAPPER_ACCEPT_MAGIC = bytes12(keccak256("WRAPPER_ACCEPT_MAGIC"));
```

Comment:
I'd make that public constant.

Reply:
Not sure. I see multiple `private constant` and `public constant` across the repo with no clear decision criteria.
In this case, the only reason I see to put it is so other contracts can be aware of the value, however, if they're implementing the `_WRAPPER_ACCEPT_MAGIC`, they should already know its value beforehand.

The magic value is not standard afaik so there's no discovery use case in which a contract _figures out_ there's a magic value, is it?

Reply:
IMO making it public contributes to discoverability (by users in etherscan, by frontend). This is particularly when the value is not standard. 

For example, in `AccessControl`, `DEFAULT_ADMIN_ROLE` is a public constant. Same for the roles in `TimelockController`.

You could imagine trying to call this, and detect if the contract is a wrapper based on the result (does it revert?)

Reply:
I think what @Amxx meant by "does it revert?" was that in order to check if a contract is a wrapper you can call `WRAPPER_ACCEPT_MAGIC()` and check if it reverts: if it doesn't revert you can assume that it's a wrapper.

I don't think we need to worry about discoverability of magic values as a general problem.

It's good to make the value publicly accessible as a convenience and that's all we need to care about at this point IMO.